install-and-configure/install/etl-backup/query-service-replicas.md

```diff
@@ -4,26 +4,28 @@
+The query service replica (QSR) is a scale-out query service that reduces load on the cost-model pod. It allows for improved horizontal scaling by being able to handle queries for larger intervals, and multiple simultaneous queries.
 
 ## Overview
 
-The query service will forward `/model/allocation` and `/model/assets` requests to a Deployment of Query Services and managed by a Load Balancer.
+The query service will forward `/model/allocation` and `/model/assets` requests to the Query Services StatefulSet.
 
 The diagram below demonstrates the backing architecture of this query service and its functionality.
 
 <figure><img src="../../../.gitbook/assets/image (5).png" alt=""><figcaption></figcaption></figure>
 
-## Prerequisites
+## Data store
 
-In order to make use of QSRs, you must first have enabled ETL backups. As of v1.100+ of Kubecost, this is enabled by default if you enable Thanos. To learn more about ETL backups, see the [ETL Backup](https://docs.kubecost.com/install-and-configure/install/etl-backup) doc.
+There are three methods to implement QSR. For environments that have Kubecost [Federated ETL](https://docs.kubecost.com/install-and-configure/install/multi-cluster/federated-etl) enabled, this store should be used.
 
-## Enabling QSR
+Alternatively, an object-store containing the ETL dataset to be queried can be configured using a secret `kubecostDeployment.queryServiceConfigSecret`. The file name of the secret must be `object-store.yaml`. Examples can be found [here](https://docs.kubecost.com/install-and-configure/install/multi-cluster/thanos-setup/long-term-storage#step-1-create-object-store.yaml).
+
+For all other environments, QSR will target the ETL backup store. As of Kubecost v1.100+, this is enabled by default if you enable Thanos. To learn more about ETL backups, see the [ETL Backup](https://docs.kubecost.com/install-and-configure/install/etl-backup) doc.
 
-After following the doc above and enabling ETL backup, you should already have set a value for the Helm flag `.Values.kubecostModel.etlBucketConfigSecret`.
+## Enabling QSR
 
-Next, set `kubecostDeployment.queryServiceReplicas` to a non-zero value. Perform a Helm upgrade with your updated _values.yaml._
+Once the data store is configured, set `kubecostDeployment.queryServiceReplicas` to a non-zero value and perform a Helm upgrade with your updated values.
 
 ## Usage
 
-Once ETL backups and the QSR have been enabled, the process will automatically apply to all Allocations or Assets queries. You can make these queries using either the Kubecost UI dashboards or direct API endpoints.
+Once QSR have been enabled, the new pods will automatically handle all API requests to `/model/allocation` and `/model/assets`.
```

Comment:
"Once QSR **has** been enabled..." If we are using the term as a plural, it should be QSRs.